remotes/docker/config/hosts.go

```diff
@@ -83,26 +83,22 @@ func ConfigureHosts(ctx context.Context, options HostOptions) docker.RegistryHos
 		}
 
 		// If hosts was not set, add a default host
-		// NOTE: Check nil here and not empty, the host may be
-		// intentionally configured to not have any endpoints
-		if hosts == nil {
-			hosts = make([]hostConfig, 1)
-		}
-		if len(hosts) > 1 {
-			if hosts[len(hosts)-1].host == "" {
-				if host == "docker.io" {
-					hosts[len(hosts)-1].scheme = "https"
-					hosts[len(hosts)-1].host = "https://registry-1.docker.io"
-				} else {
-					hosts[len(hosts)-1].host = host
-					if options.DefaultScheme != "" {
-						hosts[len(hosts)-1].scheme = options.DefaultScheme
-					} else {
-						hosts[len(hosts)-1].scheme = "https"
-					}
-				}
-				hosts[len(hosts)-1].path = "/v2"
+		if len(hosts) == 0 {
```

Comment:
The comment is not being addressed and I probably should have added a comment related to `hosts[len(hosts)-1].host == ""`. A configuration may be intentionally blocking the namespace by returning 0, that is why I am differentiating between nil and 0. If there is a bug, then maybe we need to make sure it is nil or find a better way to handle that case.  Additionally, the configuration may not be configuring the host, that is why it checks to see if the last element is not set. The configuration would only ever have the last host empty because using the Docker-compatible cert files only one entry is set (with no host), or using the config file with defaults but not setting a host override.

Reply:
OK. will change back with  `> 1` to `>=1`.

Reply:
before this commit, the empty host doesn't work because the `make([]hostConfig, 1)` has one element which never meet `>1` condition.